### PR TITLE
docs: fix README links

### PR DESCRIPTION
**Pull request by kidonng**
### Description

Fix some links in README which are either broken or redirected.

If anyone has better place to link to, suggestions welcome.

### Checklist

- [ ] Change is backwards compatible.

- [ ] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [ ] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,19 +109,20 @@ License
 
 This project is licensed under the terms of the [MIT license](LICENSE).
 
-[Nix]: https://nixos.org/nix/
+[Nix]: https://nixos.org/guides/how-nix-works.html
 [NixOS]: https://nixos.org/
-[Nixpkgs]: https://nixos.org/nixpkgs/
-[manual]: https://nix-community.github.io/home-manager/
+[Nixpkgs]: https://github.com/NixOS/nixpkgs
+[manual]: https://nix-community.github.io/home-manager/index.html
```
Comment (sumnerevans) [inline README.md:115]: this link isn't broken. https://nix-community.github.io/home-manager/ serves the `index.html` page

**Review by sumnerevans — CHANGES_REQUESTED** [review]
Not sure about the changes to the `Nix` and `Nixpkgs` links either. I don't remember what those links used to have.

File: README.md (hunk 1 above)
Comment (kidonng) [reply README.md:115]: This avoids a unnecessary refresh when you click the links in table of contents which explicitly uses `/index.html`. This is also the case for the manual links below.

**Comment by kidonng** [comment]
> Not sure about the changes to the `Nix` and `Nixpkgs` links either. I don't remember what those links used to have.

The last present version of [Nix](https://web.archive.org/web/20200426034512/https://nixos.org/nix/) and [Nixpkgs](https://web.archive.org/web/20200424062842/https://nixos.org/nixpkgs/) pages are found on Wayback Machine. The source files of these pages are removed in https://github.com/NixOS/nixos-homepage/pull/453, as a result of https://github.com/NixOS/nixos-homepage/pull/425.

From my understanding:

- The little info on Nix page is nuked. The Nixpkgs repository README indeed does not contain all that info, but I wonder if it is that useful for people who aren't familiar with Nix/NixOS.
- The info on Nix page is consolidated into the index page and "Features" page (which is now the "Explore" page). Since NixOS is already pointing to the index page, the "Explore" page seems to be the only destination what Nix can refer to, just my two cents.

**Pull request merged**